web/src/screens/public/LoginPage.js

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { compose } from 'recompose';
+import { graphql } from 'react-apollo';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+import { withStyles } from 'material-ui/styles';
+import TextField from 'material-ui/TextField';
+import Button from 'material-ui/Button';
+import Snackbar from 'material-ui/Snackbar';
+import Grid from 'material-ui/Grid';
+import Typography from 'material-ui/Typography';
+
+import { uuidv4 } from '../../utils';
+import { setCurrentUser } from '../../actions/auth';
+
+import LOGIN_MUTATION from '../../graphql/login.mutation';
+
+const styles = theme => ({
+  container: {
+    display: 'flex',
+    flexWrap: 'wrap',
+  },
+  textField: {
+    marginLeft: theme.spacing.unit,
+    marginRight: theme.spacing.unit,
+    width: 200,
+  },
+  button: {
+    margin: theme.spacing.unit,
+  },
+});
+
+class LoginPage extends React.Component {
+  state = {
+    open: false,
+    message: '',
+    username: '',
+    password: '',
+  };
+
+  onUsernameChange = (e) => {
+    const username = e.target.value;
+    this.setState(() => ({ username }));
+  };
+  onPasswordChange = (e) => {
+    const password = e.target.value;
+    this.setState(() => ({ password }));
+  };
+
+  onSubmit = (e) => {
+    e.preventDefault();
+    const deviceUuid = uuidv4();
```

Comment:
hm this brings about an interesting discussion. Should we treat browsers as devices? I think yes. We could even send push notifications to chrome and/or safari using the same frameworks as iOS and/or Android respectively.

Reply:
I agree - yes